gammapy/cube/fit.py

```diff
@@ -346,6 +346,49 @@ def create(
             gti=gti,
         )
 
+    def stack(self, other):
+        """
+        Stack the maps in other onto self.
+        The stacking is done inplace
+
+        Parameters
+        ----------
+        other: `~gammapy.cube.MapDataset`
+            the MapDatasets to be stacked with this one.
+
+        Returns
+        -------
+        new : `~gammapy.cube.MapDataset`
+            A dataset with the stacked maps
+
+        """
+
+        if self.counts:
+            if other.mask_fit:
+                other.counts = other.counts * other.mask_fit
+            self.counts.coadd(other.counts)
+
+        self.exposure.coadd(other.exposure)
+
+        if other.mask_fit:
+            other.background_model.map = other.background_model.map * other.mask_fit
+        self.background_model.map.coadd(other.background_model.evaluate())
```

Comment:
Same comment as above, maybe we should add a `weights` argument to `Map.coadd()`.